docs/migration/cosmoshub-3.md

```diff
@@ -1 +1,268 @@
+
+## Upgrade Procedure
+
+__Note__: It is assumed you are currently operating a full-node running gaia v2.0.15 with v0.37.15 of the _Cosmos SDK_.
+
+The version/commit hash of Gaia v2.0.15: `89cf7e6fc166eaabf47ad2755c443d455feda02e`
+
+1. Verify you are currently running the correct version (v2.0.15) of _gaiad_:
+
+   ```bash
+    $ gaiad version --long
+    name: gaia
+    server_name: gaiad
+    client_name: gaiacli
+    version: 2.0.15
+    commit: 89cf7e6fc166eaabf47ad2755c443d455feda02e
+    build_tags: netgo,ledger
+    go: go version go1.15 darwin/amd64
+   ```
+
+1. Make sure your chain halts at the right time and date:
+    February 18, 2021 at 06:00 UTC is in UNIX seconds: `1613628000`
+
+    ```bash
+    perl -i -pe 's/^halt-time =.*/halt-time = 1613628000/' ~/.gaiad/config/app.toml
+    ```
+
+ 1. After the chain has halted, make a backup of your `.gaiad` directory
+
+    ```bash
+    mv ~/.gaiad ./gaiad_backup
+    ```
+
+    **NOTE**: It is recommended for validators and operators to take a full data snapshot at the export
+   height before proceeding in case the upgrade does not go as planned or if not enough voting power
+   comes online in a sufficient and agreed upon amount of time. In such a case, the chain will fallback
+   to continue operating `cosmoshub-3`. See [Recovery](#recovery) for details on how to proceed.
+
+1. Export existing state from `cosmoshub-3`:
+
+   Before exporting state via the following command, the `gaiad` binary must be stopped!
+   As a validator, you can see the last block height created in the 
+   `~/.gaiad/config/data/priv_validator_state.json` - or now residing in `gaiad_backup` when you made
+    a backup as in the last step - and obtain it with
+
+   ```bash
+   cat ~/.gaiad/config/data/priv_validator_state.json | jq '.height'
+   ```
+
+   ```bash
+   $ gaiad export --for-zero-height --height=<height> > cosmoshub_3_genesis_export.json
+   ```
+   _this might take a while, you can expect an hour for this step_
+
+1. Verify the SHA256 of the (sorted) exported genesis file:
+
+    Compare this value with other validators / full node operators of the network. 
+    Going forward it will be important that all parties can create the same genesis file export.
+
+   ```bash
+   $ jq -S -c -M '' cosmoshub_3_genesis_export.json | shasum -a 256
+   [SHA256_VALUE]  cosmoshub_3_genesis_export.json
+   ```
+
+1. At this point you now have a valid exported genesis state! All further steps now require
+v4.0.0 of [Gaia](https://github.com/cosmos/gaia). 
+Cross check your genesis hash with other peers (other validators) in the chat rooms.
+
+   **NOTE**: Go [1.15+](https://golang.org/dl/) is required!
+
+   ```bash
+   $ git clone https://github.com/cosmos/gaia.git && cd gaia && git checkout v4.0.0; make install
+   ```
+
+1. Verify you are currently running the correct version (v4.0.0) of the _Gaia_:
+
+    The upgrade time as agreed upon by governance: 
```

Comment:
Should this go below the next block? Looks like the version block should be here and the time underneath